profiling/src/internal/profile.rs

```diff
@@ -417,9 +404,10 @@ impl Profile {
             encoder.encode(ProfileFunctionsEntry::from(item))?;
         }
 
-        for item in self.strings.into_iter() {
+        for item in self.strings.iter() {
             encoder.encode(ProfileStringTableEntry::from(item))?;
         }
+        drop(self.strings);
```

Comment:
Does this drop the memory that is backing the `'a &str` in the `ProfileStringTableEntry`?

Reply:
Miri didn't complain and I don't think so, but I haven't stepped through the code to be sure.

Reply:
`encode` calls `encode_raw` which calls `put_slice` which does the copy, so I'm pretty confident this is fine.